packages/artillery/lib/util/validate-script.js

```diff
@@ -105,9 +105,13 @@ const scenarioItem = Joi.object({
 });
 
 const beforeAfterSchema = Joi.object({
```

Comment:
Previously the schema would look at `config.engines` to determine this. However, this is more complicated than just looking at the `engine` field within before/after itself:
- If it doesn't have it, it's http (default), so it needs flow.
- If it has it, if it's `socketio`, `ws` or `http`, it needs flow. Otherwise it does not require flow (e.g. custom engine or Playwright)